comit/src/network/protocols/orderbook.rs

```diff
@@ -72,20 +72,27 @@ impl Orderbook {
             config,
         );
 
-        let mut orderbook = Orderbook {
+        let orderbook = Orderbook {
             peer_id,
             gossipsub,
             take_order: behaviour,
             orders: HashMap::new(),
             events: VecDeque::new(),
         };
 
-        // Since we only support a single trading pair topic just subscribe to it now.
         orderbook
-            .gossipsub
-            .subscribe(Topic::new(BTC_DAI.to_string()));
+    }
 
-        orderbook
+    /// Subscribe to the BTC/DAI gossipsub topic. Caller is responsible for
+    /// ensuring that there is an open connection to another node in the network
+    /// before calling.
+    // Currently we call subscribe after dialing, this implies that in order to
+    // receive orders on the gossipsub network one must dial to another node. For a
+    // node acting as a maker only it probably does not matter if we do not receive
```

Comment:
I believe @bonomat had a strong opinion on the subject that all takers and makers must participate to the gossip network.
I understand this workaround is temporary and being worked on upstream, right?

Reply:
It is my opinion that we need a sit down with the folks invested in the orderbook to discuss the design, the goals, and what not. @bonomat, @rishflab, and myself have a meeting today about it but that does not include anyone from Nectar. I personally do not know why the orderbook was designed the way it is, I do not fully understand what the goal is with it, with Nectar, or with cnd's usage. Perhaps its just me?

Reply:
We actually inadvertly started the discussion on the mailing so let's see how it goes there.